qiskit/circuit/qpy_serialization.py

```diff
@@ -160,6 +160,93 @@
 which is immediately followed by ``pauli_op_size`` bytes which are .npy format [#f2]_
 data which represents the :class:`~qiskit.quantum_info.SparsePauliOp`.
 
+Version 3 of the QPY format also defines a struct format to represent a
+:class:`~qiskit.circuit.ParameterVectorElement` as a distinct subclass from
+a :class:`~qiskit.circuit.Parameter`. This adds a new parameter type char ``'v'``
+to represent a :class:`~qiskit.circuit.ParameterVectorElement` which is now
+supported as a type string value for an INSTRUCTION_PARAM. The payload for these
+parameters are defined below as :ref:`param_vector`.
+
+.. _param_vector:
+
+
+PARAMETER_VECTOR_ELEMENT
+------------------------
+
+A PARAMETER_VECTOR_ELEMENT represents a :class:`~qiskit.circuit.ParameterVectorElement`
+object the data for a INSTRUCTION_PARAM. The contents of the PARAMETER_VECTOR_ELEMENT are
+defined as:
+
+.. code-block:: c
+
+    struct {
+        uint16_t vector_name_size;
+        uint64_t vector_size;
+        char uuid[16];
+        uint64_t index;
+    }
+
+which is immediately followed by ``vector_name_size`` utf8 bytes representing
+the parameter's vector name.
+
+.. _param_expr_v3:
+
+
+PARAMETER_EXPR
+--------------
+
+Additionally, since QPY format version v3 distinguishes between a
+:class:`~qiskit.circuit.Parameter` and :class:`~qiskit.circuit.ParameterVectorElement`
+the payload for a :class:`~qiskit.circuit.ParameterExpression` needs to be updated
+to distinguish between the types. The following is the modified payload format
+which is mostly identical to the format in Version 1 and :ref:`version_2` but just
+modifies the ``map_elements`` struct to include a symbol type field
+
+
+A PARAMETER_EXPR represents a :class:`~qiskit.circuit.ParameterExpression`
+object that the data for an INSTRUCTION_PARAM. The contents of a PARAMETER_EXPR
+are defined as:
+
+The PARAMETER_EXPR data starts with a header:
```

Comment:
This looks like a bad merge.

Reply:
Heh, more just sloppy copy and pasting. I copied the full v1 doc into this section on v3 after writing that the blurb on L206-L208. But yeah this looks awkward we should fix it.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/7381/commits/cbd45f855aca3cfd3302f61017c62fc74df60a31